plugin/index.js

```diff
@@ -49,89 +50,66 @@ function getBundleMetrics(stats, inspectpack, handler, done) {
+        code: bundle.source,
+        root: bundle.context,
+        format: "object",
+        minified: true,
+        gzip: true
+      })
+        .then(metrics => ({
+          path: bundle.path,
+          metrics
+        }))
+    )
+  )
+    .then(bundle => ({
+      type: "sizes",
+      value: bundle
+    }))
+    .catch(err => ({
+      type: "sizes",
+      error: true,
+      value: serializeError(err)
+    }));
+
+  const getProblems = bundles => Promise.all(
+    INSPECTPACK_PROBLEM_ACTIONS.map(action =>
+      Promise.all(
+        bundles.map(bundle =>
+          inspectpack[action]({
             code: bundle.source,
             root: bundle.context,
+            duplicates: true,
             format: "object",
             minified: true,
             gzip: true
           })
-          .then(metrics => ({
-            path: bundle.path,
-            metrics
-          }))
-      )
-    )
-      .then(bundle => {
-        handler([
-          {
-            type: "sizes",
-            value: bundle
-          }
-        ]);
-        return true;
-      })
-      .catch(err => {
-        handler([
-          {
-            type: "sizes",
-            error: true,
-            value: serializeError(err)
-          }
-        ]);
-      }),
-    Promise.all(
-      INSPECTPACK_PROBLEM_ACTIONS.map(action =>
-        Promise.all(
-          bundles.map(bundle =>
-            inspectpack
-              [action]({
-                code: bundle.source,
-                root: bundle.context,
-                duplicates: true,
-                format: "object",
-                minified: true,
-                gzip: true
-              })
-              .then(metrics => ({
-                path: bundle.path,
-                [action]: metrics
-              }))
-          )
+            .then(metrics => ({
+              path: bundle.path,
+              [action]: metrics
+            }))
         )
       )
     )
-      .then(bundle => {
-        handler([
-          {
-            type: INSPECTPACK_PROBLEM_TYPE,
-            error: true,
-            value: _.flatten(bundle)
-          }
-        ]);
-        return true;
-      })
-      .catch(err => {
-        handler([
-          {
-            type: INSPECTPACK_PROBLEM_TYPE,
-            error: true,
-            value: serializeError(err)
-          }
-        ]);
-      })
-  ])
-    .then(() => {
-      done();
-    })
-    .catch(err => {
-      console.error("INSPECTPACK ERROR: ", err);
-      done();
-    });
+  )
+    .then(bundle => ({
+      type: INSPECTPACK_PROBLEM_TYPE,
+      value: _.flatten(bundle)
+    }))
+    .catch(err => ({
+      type: INSPECTPACK_PROBLEM_TYPE,
+      error: true,
+      value: serializeError(err)
+    }));
+
+  const sizesStream = most.of(bundlesToObserve).map(getSizes);
+  const problemsStream = most.of(bundlesToObserve).map(getProblems);
+
+  return most.mergeArray([sizesStream, problemsStream])
+    .chain(most.fromPromise);
```

Comment:
This is like `most.awaitPromises()` except it preserves arrival order. See https://github.com/cujojs/most/blob/master/docs/api.md#awaitpromises